components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/handlers/SCIMClaimOperationEventHandler.java

```diff
@@ -104,4 +119,37 @@ public String getName() {
 
         return "SCIMClaimOperationEventHandler";
     }
+
+    private void handleSCIMExternalClaimAddEvent(Event event) {
+
+        if (!IdentityEventConstants.Event.PRE_ADD_EXTERNAL_CLAIM.equals(event.getEventName())) {
+            return;
+        }
+
+        String claimDialectUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.CLAIM_DIALECT_URI);
+        String externalClaimUri = (String) event.getEventProperties()
+                .get(IdentityEventConstants.EventProperty.EXTERNAL_CLAIM_URI);
+        Set<String> scimClaimDialects = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+                SCIM_CORE_CLAIM_DIALECT,
+                SCIM_USER_CLAIM_DIALECT,
+                SCIM_ENTERPRISE_USER_CLAIM_DIALECT,
+                SCIM_SYSTEM_USER_CLAIM_DIALECT
+        )));
+
+        /*
+         * All spec-defined claims might not be added to dialects as external claims. All supported claims can be found
+         * through the schemas.profile config defined in the identity.xml. The DialectConfigParser is used to read the
+         * final server supported claims.
+         */
+        if (DialectConfigParser.getInstance().getClaimsMap().get(externalClaimUri) != null
+                && DialectConfigParser.getInstance().getClaimsMap().get(externalClaimUri).equals(claimDialectUri)) {
+            return;
+        }
+
+        if (scimClaimDialects.contains(claimDialectUri)) {
+            IdentityUtil.threadLocalProperties.get()
```

Comment:
Let's add a comment mentioning the purpose. This thread local is added to communicate that a violation has been done.

Reply:
Fixed with 5af70228808cebfd3bc272179fce9a76e586830f